Bug 1519323 - Convert revisionInformation to ReactJS

ionutgoldan opened the pull request.

sarah-clements commented on ui/perfherder/RevisionInformation.jsx:
```diff
@@ -0,0 +1,78 @@
+export default class RevisionInformation extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {};
```
Since state is not used, this can be made a functional component.

sarah-clements commented on ui/perfherder/RevisionInformation.jsx:
```diff
@@ -0,0 +1,78 @@
+      <ul className="list-inline push-information">
+        {originalRevision && (
+          <li className="list-inline-item">
+            <strong>Base</strong> -&nbsp;
```
Since the content of each list item is just a formatted string, I wonder if there's an easy way to use a template literal or formatting function.

sarah-clements commented on ui/perfherder/helpers.js:
```diff
@@ -807,3 +812,6 @@ export const nudgeAlert = (dataPoint, towardsDataPoint) => {
 };
+
+export const getRevisionUrl = (revision, projectName) =>
+  revision ? getJobsUrl({ repo: projectName, revision }) : '';
```
It might be better to return plain text rather than an empty anchor tag, since if it's still clickable users might think it's broken (you could probably do this in render instead unless you know this will be used elsewhere).

ionutgoldan commented:
> 
> 
> This looks good. A few things that I may not have mentioned:
> 
>     * We're replacing bootstrap with react-strap so be on the lookout for any CSS classes that pertain
>       to bootstrap js functionality; they'll need to be replaced with react-strap components. Up to you if you want to change the List Items to components.
> 
>     * We're also favoring use of bootstrap utility classes for things like spacing, alignment, etc. It's likely some custom CSS used for Perfherder can be removed and replaced with bootstrap utilities.

Good point and thanks! I've addressed these requests as well.

ionutgoldan commented:
@sarah-clements could you have a second look over this PR?

sarah-clements commented on ui/perfherder/RevisionInformation.jsx:
```diff
@@ -0,0 +1,101 @@
+  return (
+    <React.Fragment>
+      <strong>{baselineOrNew}</strong> -&nbsp;
+      {revisionUrl ? (
```
The `revisionUrl` will always be truthy since `getRevisionUrl` is either returning a hyperlink or an empty string. You could get rid of `getRevisionUrl` entirely and move that logic into here, only generating the url if revision is true.

sarah-clements commented on ui/js/filters.js:
```diff
@@ -2,17 +2,12 @@
-        }
-        return '';
-    };
+    return getRevisionUrl;
```
Can make this a one-liner with the arrow function.

edmorley replied:
Or even:

```js
treeherder.filter('getRevisionUrl', getRevisionUrl);
```

edmorley commented on ui/perfherder/helpers.js:
```diff
@@ -754,3 +759,6 @@ export const nudgeAlert = (dataPoint, towardsDataPoint) => {
   return update(getApiUrl(`/performance/alert/${alertId}/`), towardsDataPoint);
 };
+
+export const getRevisionUrl = (revision, projectName) =>
```
I think this is general enough to belong in `ui/helpers/url.js` rather than a perfherder-only helper?

edmorley commented on ui/perfherder/RevisionInformation.jsx:
```diff
@@ -0,0 +1,101 @@
+  const revisionUrl = getRevisionUrl(revision, project.name);
+  const truncatedRevision = revision.substring(0, 12);
+  let baselineOrNew = isBaseline ? 'Base' : 'New';
+  if (selectedTimeRange) baselineOrNew = 'Base';
```
Another way of writing these two lines might be:

```js
let baselineOrNew = isBaseline || selectedTimeRange ? 'Base' : 'New';
```

ionutgoldan replied:
Nifty!

ionutgoldan commented:
@sarah-clements could you have a second look over this PR?

> 
> 
> Left a few more comments but otherwise looks good. In the future, can you squash as many of your commits as possible when opening a pull request, and then keep your changes to one commit per round of changes with a title like "changes based on feedback"? It makes it easier to review changes.

Sure, will do this from now on.

sarah-clements commented:
@ionutgoldan We had to do a revert of your commit due to an error in RevisionInformation component causing that component to not render on stage (due to this merge). The console showed 'Revision is undefined' in the getRevisionSpecificDetails component along with some propType errors. Can you fix these errors and open a pull request?
